Restrict new tests to recent versions of PETSc.

jwpeterson opened the pull request:
We need to restrict the new Nemesis elemental variable tests since
they were generated using PETSc 3.8.0, and therefore used a version of
METIS/ParMETIS with an updated, architecture-independent random number
generator. These tests _should_ be platform-independent, although that
of course needs to be verified.

Refs libMesh/libmesh#1764.

jwpeterson committed:
Restrict new tests to recent versions of PETSc.

We need to restrict the new Nemesis elemental variable tests since
they were generated using PETSc 3.8.0, and therefore used a version of
METIS/ParMETIS with an updated, architecture-independent random number
generator. These tests _should_ be platform-independent, although that
of course needs to be verified.

Refs libMesh/libmesh#1764.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/206784/) on e39ebc6 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11776/site)

This comment will be updated on new commits.

jwpeterson committed:
Add missing gold files.

These files weren't added because they are ignored by default, we
allow gold/*.e but not gold/*.e.?.?

Refs libMesh/libmesh#1764.

permcody commented:
You could modify the .gitignore file:
https://github.com/idaholab/moose/blob/devel/.gitignore#L90-L96
